engine/runtime-compiler/src/main/java/org/enso/compiler/pass/analyse/types/BuiltinTypes.java

```diff
@@ -0,0 +1,33 @@
+package org.enso.compiler.pass.analyse.types;
+
+import org.enso.pkg.QualifiedName;
+import org.enso.pkg.QualifiedName$;
+
+/** A helper class providing the builtin types. */
+public class BuiltinTypes {
+  // TODO in next iterations we will want to resolve descriptions of these types based on the loaded
+  // std-lib (from PackageRepository, if available)
```

Comment:
Thanks for the heads-up, I did not think about that case - that is going to be interesting...

I think I will leave this problem for now, as the issue is not apparent with the current setup. I will attempt to handle this around Iteration 2, when I will be actually resolving methods on these types - then I will want to correlate my builtin types with the IR/Bindings for their definitions - so this will be the right place to see if the stdlib is imported or not and use the full IR or the limited builtin mode instead.